src/login/adal/DeviceTokenCredentials2.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AccessToken, TokenCredential } from "@azure/core-auth";
+import { Constants as MSRestConstants, WebResource } from "@azure/ms-rest-js";
+import { DeviceTokenCredentials } from '@azure/ms-rest-nodeauth';
+import { TokenResponse } from "adal-node";
+
+/**
+ * Token that is forward compatible with track 2 Azure SDK for Node.js
+ * `getToken: Promise<AccessToken>` is required for use with T2 Azure SDK, but doesn't
+ * affect T1 SDKs as those require `signRequest`
+ * `DeviceTokenCredentials` forces `getToken` to return `TokenResponse` so this is to
+ * overwrite that implementation
+ */
+export class DeviceTokenCredentials2 extends DeviceTokenCredentials implements TokenCredential {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	public async getToken(): Promise<any> {
+		const tokenResponse = <AccessToken & TokenResponse>(await this.getTokenFromCache(this.username));
+		return Object.assign(tokenResponse, {token: tokenResponse.accessToken, expiresOnTimestamp: tokenResponse.expiresIn});
+	}
+
+	public async signRequest(webResource: WebResource): Promise<WebResource> {
+		const tokenResponse: AccessToken = <AccessToken>(await this.getToken());
+			webResource.headers.set(
+				MSRestConstants.HeaderConstants.AUTHORIZATION,
+				`${MSRestConstants.HeaderConstants.AUTHORIZATION_SCHEME} ${tokenResponse.token}`
```

Comment:
I don't think we need this function because we're only adding to the return type of `this.getToken()`, not removing anything that used to be there. If that's the case we can rely on the existing implementation of `signRequest` from the parent class. But like I said I think I may be misunderstanding something...

Reply:
I think this will work now that we are implementing `TokenResponse & AccessToken`.  Originally, I only was returning an `AccessToken` so of course `signRequest` was failing because it uses `TokenResponse` to build the request.